zcash_client_backend/src/decrypt.rs

```diff
@@ -49,21 +51,29 @@ pub fn decrypt_transaction(
         };
 
         for (account, (ivk, ovk)) in vks.iter().enumerate() {
-            let ((note, to, memo), outgoing) =
-                match try_sapling_note_decryption(ivk, &epk, &output.cmu, &output.enc_ciphertext) {
-                    Some(ret) => (ret, false),
-                    None => match try_sapling_output_recovery(
-                        ovk,
-                        &output.cv,
-                        &output.cmu,
-                        &epk,
-                        &output.enc_ciphertext,
-                        &output.out_ciphertext,
-                    ) {
-                        Some(ret) => (ret, true),
-                        None => continue,
-                    },
-                };
+            let ((note, to, memo), outgoing) = match try_sapling_note_decryption(
+                parameters,
+                tx.expiry_height,
```

Comment:
should we be using `tx.expiry_height` for the plaintext version check?

Reply:
This will only work for transactions that set an expiry height (`0` means "no expiry", which is something we should represent within the expiry height type).

We should be using the same value that is used in `zcashd` when trial-decrypting, which is the height passed to `SyncWithWallets` in `ThreadNotifyWallets` (the block height for mined transactions, and the "mempool height" (chain height + 1) for mempool transactions. For this API, have the caller pass it in.